integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/utils/DataSourceV2RelationDatasetExtractor.java

```diff
@@ -31,16 +32,39 @@
 @Slf4j
 public class DataSourceV2RelationDatasetExtractor {
 
+  public static <D extends OpenLineage.Dataset> List<D> extract(
+      DatasetFactory<D> datasetFactory,
+      OpenLineageContext context,
+      DataSourceV2Relation relation,
+      boolean includeVersionFacet) {
```

Comment:
Looks good overall, wdyt about splitting extract method into two separate methods - `extract` and `extractIncludingVersionFace`t?